### PR TITLE
Hide "collection from" text when a location has an incomplete address.

### LINKED ISSUE #9789: Local Pickup Title shows twice
## Describe the bug
In #8305 the string "Collection from" was added.

Thats a nice addition but leads together with the Title field in the local pickup option to a duplicated text. I had entered here a text (required field) and translated "Local Pickup" to a german Word "Abholung vor Ort"
image

In the Location i did set a Location but no Adress - just the name and pickup Details
image

Then it pretty much showed the same text duplicated "Collection from Local Pickup" (example in english) and the name of the the store.
image

PS: Same already is visible below the Shipping title...

## To reproduce

1. Go to /wp-admin/admin.php?page=wc-settings&tab=shipping&section=pickup_location
2. Click on Enable Local Pickup & set a title "Local Pickup"
3. Add a Pickup Location and set only Location Name and Pickup Details
4. See the cart and the doubled text "Collection from Local Pickup (Name of Store)"

## Versions
WooCommerce 7.7.2 & WooBlocks 10.4.0

## Expected behavior
If no pickup adress is set then dont show the "Collection from" string.

## Screenshots
see above

### PR DESCRIPTION
**Pull request by mikejolley**
Hides the "collection from <address>" section when a local pickup location has no defined address. This prevents repetition of the local pickup rate name, or nonsensical text. 

Fixes #9789

#### User Facing Testing

1. Go to WooCommerce > Settings > Shipping > Local Pickup. Create 2 pickup locations—one should have a full address/pickup location, and one should have no address like so:

Screenshot 2023-06-13 at 12 48 21

2. Add items to your cart and go to checkout.
3. Select Local Pickup
4. Select the pickup location with an address. it should show "collection from" like so:

Screenshot 2023-06-13 at 12 50 13

5. Select the pickup location with no address. It should not show the collection from section:
 
Screenshot 2023-06-13 at 12 50 19

6. Place the order and check the confirmation displays correctly without an address also:

Screenshot 2023-06-13 at 12 56 19

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Local Pickup - Only show collection address when set.

### CODE REVIEW

**Review by tarunvijwani — COMMENTED** [review]
This isn't working for me. Am I missing anything?

https://github.com/woocommerce/woocommerce-blocks/assets/11503784/da89ee54-ca5e-402f-b947-f9ba72a9d6fc

**Comment by mikejolley** [comment]
@tarunvijwani thats the trunk behaviour. I just retested and its fine in this branch. Suggest rebuilding and clearing caches, or someone else from @woocommerce/rubik can try it.

**Comment by nielslange** [comment]
> @tarunvijwani thats the trunk behaviour. I just retested and its fine in this branch. Suggest rebuilding and clearing caches, or someone else from @woocommerce/rubik can try it.

@mikejolley I'm getting the same result as @tarunvijwani. To ensure that I'm starting with a fresh environment, I ran the following commands:

```sh
rm composer.lock
rm package-lock.json
rm -rf node_modules
rm -rf vendor
nvm use
composer i
npm i
npm run build
```

<table>
<tr>
<td valign="top">Local Pickup address:
<br><br>

Screenshot 2023-06-18 at 17 52 06
</td>
<td valign="top">Cart block:
<br><br>

Screenshot 2023-06-18 at 17 53 29
</td>
<td valign="top">Checkout block:
<br><br>

Screenshot 2023-06-18 at 17 54 12
</td>
</tr>
</table>

File: src/Shipping/PickupLocation.php [hunk 1]
```diff
@@ -42,6 +56,7 @@ public function calculate_shipping( $package = array() ) {
 				if ( ! $location['enabled'] ) {
 					continue;
 				}
+				$has_pickup_address = ! empty( $location['address'] ) && ! empty( $location['address']['city'] ) && ! empty( $location['address']['state'] ) && ! empty( $location['address']['country'] );
```
Comment (tarunvijwani) [inline src/Shipping/PickupLocation.php]: Some countries like UAE don't have a state. This condition will hide the `Collection from` for them as well. 
I feel we should have helper function that will check the address completeness based on the country. 
We have one similar function: `isAddressComplete` in utils. Do you think we should utilize it here?
You can test that with the below address:
image

**Review by tarunvijwani — CHANGES_REQUESTED** [review]
You were right! I was testing it on WooCommerce 7.9. Forgot to enable the WC Blocks plugin after the release.😃
It works as expected. Great work! 🎉 
I left one suggestion for the countries where state is an optional field. Let me know your thoughts about it.

**Review by opr — CHANGES_REQUESTED** [review]
Code wise it looks great, Tarun's suggestion is valid though! We should avoid checking for fields that may are not required for that locale.

What do you think of this?

```php
$required_address_fields_for_country = [];
if ( ! empty( $location['address']['country'] ) ) {
	$shipping_address_fields = wc()->countries->get_address_fields( $location['address']['country'], 'shipping_' );
	foreach ( $shipping_address_fields as $field_name => $field ) {
		if ( true === $field['required'] ) {
			$required_address_fields_for_country[] = str_replace( 'shipping_', '', $field_name );
		}
	}
}

$has_pickup_address = ! empty( $location['address'] ) &&
	( ! empty( $location['address']['city'] ) || ! in_array( 'city', $required_address_fields_for_country, true ) ) &&
	( ! empty( $location['address']['state'] ) || ! in_array( 'state', $required_address_fields_for_country, true ) ) &&
	( ! empty( $location['address']['country'] ) );
```

**Comment by mikejolley** [comment]
I added a helper in https://github.com/woocommerce/woocommerce-blocks/pull/9808/commits/b4bfc94fbfd58a319b13b66058388bd6dd2d8113

**Review by opr — APPROVED** [review]
That helper function is a much cleaner way of doing it than I suggested, nice one!

Do you think we should add tests for it? We could add an issue for this and write them later.

**Review by tarunvijwani — CHANGES_REQUESTED** [review]
Thank you for updating the code. It seems it is still returning null for the UAE address. 

image


The variable $country_address_fields is returning the correct data though:

| Address | country_address_fields value |
|---|---|
|image|image|

**Comment by mikejolley** [comment]
@tarunvijwani missed a check to see if the field was included in address_fields. Try now.

**Review by tarunvijwani — APPROVED** [review]
Works great! Thank you for the changes! 🎉

**Pull request merged**